summoner-cli/src/Summoner/Template/Nix.hs

```diff
@@ -0,0 +1,142 @@
+{-# LANGUAGE QuasiQuotes    #-}
+
+module Summoner.Template.Nix
+       ( nixFiles
+       ) where
+
+import NeatInterpolation (text)
+
+import Summoner.Default (defaultGHC)
+import Summoner.GhcVer (GhcVer (..))
+import Summoner.Settings (NixPkgSet (..), Settings (..), defaultNixPkgSet)
+import Summoner.Tree (TreeFs (..))
+
+import qualified Data.Text as T
+
+nixCompiler :: GhcVer -> Text
+nixCompiler = \case
+    Ghc7103 -> "ghc7103"
+    Ghc801  -> "ghc802"
+    Ghc802  -> "ghc802"
+    Ghc822  -> "ghc822"
+    Ghc843  -> "ghc844"
+    Ghc844  -> "ghc844"
+    Ghc863  -> "ghc863"
+
+defaultNixCompiler :: Text
+defaultNixCompiler = nixCompiler defaultGHC
+
+nixFiles :: Settings -> [TreeFs]
+nixFiles Settings{..} =
+    [ shellNix
+    , releaseNix
+    , defaultNix
+    , nixDir
+    ]
+    where
+        defaultNix = File "default.nix" defaultNixT
+        shellNix = File "shell.nix" shellNixT
+        releaseNix = File "release.nix" releaseNixT
+
+        nixDir = Dir "nix" [fetchNixpkgsNix, nixpkgsNix, overridesNix]
+        fetchNixpkgsNix = File "fetchNixpkgs.nix" fetchNixpkgsNixT
+        nixpkgsNix = File "nixpkgs.nix" nixpkgsNixT
+        overridesNix = File "overrides.nix" overridesNixT
+
+        defaultNixT :: Text
+        defaultNixT =
+            [text|
+            { compiler ? "${defaultNixCompiler}"
+            , nixpkgs ? (import ./nix/nixpkgs.nix { inherit compiler; })
+            }:
+
+            with rec {
+              drv = nixpkgs.haskellPackages.${settingsRepo};
+            };
+
+            drv
+            |]
+
+        shellNixT :: Text
+        shellNixT = "(import ./default.nix {}).env" <> endLine
+
+        releaseNixT :: Text
+        releaseNixT = T.concat
+            [ "{"
+            , endLine
+            , T.intercalate endLine
+                ( map
+                  (\ghcV -> let compiler = nixCompiler ghcV
+                            in compiler <> "  = import ./default.nix { compiler = \"" <> compiler <> "\";  };"
+                  )
+                  settingsTestedVersions
+                )
+            , endLine
+            , "}"
+            , endLine
+            ]
+
+        fetchNixpkgsNixT :: Text
+        fetchNixpkgsNixT = T.concat
+            [ "{ owner  # The owner of the nixpkgs", endLine
+            , ", repo   # The repo of the nixpkgs", endLine
+            , ", rev    # The Git revision of nixpkgs to fetch", endLine
+            , ", sha256 # The SHA256 hash of the unpacked archive", endLine
+            , "}:", endLine, endLine
+            , "builtins.fetchTarball {", endLine
+            , "  url = \"https://github.com/${owner}/${repo}/archive/${rev}.tar.gz\";", endLine
+            , "  inherit sha256;", endLine
+            , "}"
+            , endLine
+            ]
+
+        NixPkgSet { npsOwner, npsRepo, npsRev, npsSha } = case settingsNixPkgSet of { Just n -> n; Nothing -> defaultNixPkgSet }
+
+        nixpkgsNixT :: Text
+        nixpkgsNixT = T.concat
```

Comment:
Why not use `neat-interpolation` here to make it human-readable?

Generally, we sometimes avoid it because of the empty line it can leave depending on we have the empty text or not, but here I don't see that anything like this is present. Probably you had other reasons?